src/es-module-shims.js

```diff
@@ -97,6 +100,10 @@ function metaResolve (id, parentUrl = this.url) {
 
 importShim.resolve = resolveSync;
 importShim.getImportMap = () => JSON.parse(JSON.stringify(importMap));
+importShim.setImportMap = importMapIn => {
+  if (!shimMode) throw new Error('Unsupported in polyfill mode.');
+  importMap = JSON.parse(JSON.stringify(importMapIn));
```

Comment:
Of course I completely forgot we need to properly normalize the import map, in which case we should probably make this an `extendImportMap` function instead of `setImportMap` as well:

```suggestion
  importMap = resolveAndComposeImportMap(importMapIn, pageBaseUrl, importMap);
```